### PR TITLE
[AMSDK-11351] - EdgeIdentity and EdgeConsent API's

### PR DESCRIPTION
**Pull request by PravinPK**
- Redesigned menu screen to look like iOS's Sample app
- Added separate tabs for EdgeIdentity and Edge consent
- Included a version number card for all extensions
- Removed the Action bar for more interactable space

**Screenshot**
**Menu Screen**

Screen Shot 2021-04-09 at 3 31 36 PM

**Edge Extension**

Screen Shot 2021-04-09 at 3 32 46 PM

**EdgeIdentity Extension**

Screen Shot 2021-04-09 at 3 33 38 PM

**Consent Extension**

Screen Shot 2021-04-09 at 3 34 54 PM

### CODE REVIEW

File: Sample-App/app/build.gradle [hunk 1]
```diff
@@ -26,11 +26,14 @@ dependencies {
...
     implementation 'com.google.android.gms:play-services-location:17.1.0'
+    implementation 'com.google.code.gson:gson:2.3.1'
 
 
     implementation 'com.adobe.marketing.mobile:userprofile:1.1.0'
     implementation 'com.adobe.marketing.mobile:sdk-core:1.5.7'
     implementation 'com.adobe.marketing.mobile:assurance:1.0.1'
     implementation 'com.adobe.marketing.mobile:edge:1.+'
+    implementation 'com.adobe.marketing.mobile:edgeconsent:1.+'
+    implementation 'com.adobe.marketing.mobile:edgeidentity:1.0.0-SNAPSHOT'
```
Comment (PravinPK) [inline Sample-App/app/build.gradle:37]: will can change this to the production version once released

Comment (emdobrin) [reply Sample-App/app/build.gradle:37]: we can now

File: Sample-App/build.gradle [hunk 2]
```diff
@@ -15,7 +15,7 @@ allprojects {
     repositories {
         google()
         jcenter()
-        
+        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
```
Comment (emdobrin) [inline Sample-App/build.gradle:18]: can we remove jcenter and add maven ?

File: Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeTab.java [hunk 3]
```diff
@@ -135,6 +143,78 @@ public void onClick(View v) {
         // Setup product list spinner
         Spinner spinner = view.findViewById(R.id.products_spinner);
         spinner.setAdapter(new SpinnerAdapter(getActivity(), android.R.layout.simple_spinner_item, products));
+
+
+        // EdgeIdentity API's
```
Comment (emdobrin) [inline Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeTab.java:148]: this looks a little crowded in a single page, can we create separate pages for Edge Identity and Consent samples?

Comment (emdobrin) [reply Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeTab.java:148]: Also, can we remove the top tabs if we are using buttons on main page?

Comment (PravinPK) [reply Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeTab.java]: Created separate pages for Consent and IdentityEdge with textView to show the latest data.

We can still keep the tab... trust me is super easy to navigate once your in that page

File: Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeConsentTab.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A simple {@link Fragment} subclass.
+ * create an instance of this fragment.
+ */
+public class EdgeConsentTab extends Fragment implements NavigationAware {
+    private static final String LOG_TAG = "EdgeConsentTab";
```
Comment (emdobrin) [inline Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeConsentTab.java:35]: let's rename to ConsentTab everywhere

File: Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeConsentTab.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    }
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        // Card view to display registered Consent extension version
+        final TextView tv_version  = getView().findViewById(R.id.tv_edgeConsent_version);
+        view.post(new Runnable() {
+            @Override
+            public void run() {
+                if (tv_version != null) {
+                    tv_version.setText("Edge Consent Extension version: " + Consent.extensionVersion());
```
Comment (emdobrin) [inline Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeConsentTab.java:56]: First of all, thank you for this change!!!! 

can be simplified to "Consent v" + Consent.extensionVersion

File: Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeIdentityTab.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
+/*
+  Copyright 2020 Adobe
```
Comment (emdobrin) [inline Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeIdentityTab.java:2]: 2021 - same for all new files

File: Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeIdentityTab.java [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+    }
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        // Card view to display registered Edge Identity extension version
+        final TextView tv_version  = getView().findViewById(R.id.tv_edgeidentity_version);
+        view.post(new Runnable() {
+            @Override
+            public void run() {
+                if (tv_version != null) {
+                    tv_version.setText("Edge Identity Extension version: " + Identity.extensionVersion());
```
Comment (emdobrin) [inline Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeIdentityTab.java:56]: "Edge Identity v" + ..

File: Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/AssuranceTab.java [hunk 8]
```diff
@@ -20,42 +20,38 @@
...
     }
 
     @Override
     public void onViewCreated(View view, Bundle savedInstanceState) {
+        final TextView tv_version  = getView().findViewById(R.id.tv_assurance_version);
+        view.post(new Runnable() {
+            @Override
+            public void run() {
+                if (tv_version != null) {
+                    tv_version.setText("Assurance Extension version: " + Assurance.extensionVersion());
```
Comment (emdobrin) [inline Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/AssuranceTab.java:44]: "Assurance v" + ..

File: Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeTab.java (hunk 3 above)
Comment (emdobrin) [reply Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeTab.java]: 👍  for now it should be ok, but for the future it does not accommodate multiple extensions

**Comment by emdobrin** [comment]
Looks great after your latest changes 💯 thanks PK

**Pull request merged**